extensions/guacamole-auth-cas/src/main/java/org/apache/guacamole/auth/cas/AuthenticationProviderService.java

```diff
@@ -83,7 +106,16 @@ public AuthenticatedUser authenticateUser(Credentials credentials)
             String ticket = request.getParameter(CASTicketField.PARAMETER_NAME);
             if (ticket != null) {
                 AuthenticatedUser authenticatedUser = authenticatedUserProvider.get();
-                authenticatedUser.init(ticketService.processUsername(ticket), credentials);
+                AttributePrincipal principal = ticketService.validateTicket(ticket);
+                String username = principal.getName();
+                credentials.setUsername(username);
+                Object credObj = principal.getAttributes().get("credential");
+                if (credObj != null) {
+                    String clearPass = decryptPassword(credObj.toString());
```

Comment:
Is there a defined object type for what `get("credential")` returns? Or is conversion to a string and back again the only way?